_episodes/l2_02_basic_unit_testing.md

```diff
@@ -0,0 +1,293 @@
+- [doctest](https://github.com/doctest/doctest)
+
+In this course, we will focus mainly on GoogleTest.
+
+
+
+
+## 2. A brief about GoogleTest
+
+Simply defining, GoogleTest is a testing framework developed by Google's testing technology team to develop and write C++ tests.  GoogleTest offers multiple advantages over other frameworks:-
+
+1. **Comprehensive Features**: Google Test provides a rich set of features, including a wide range of assertion macros, test fixtures, parameterized tests, test discovery, test filtering, and powerful mocking capabilities. It offers a complete testing framework that can handle various testing scenarios.
+2. **Integration with Google Test ecosystem**: Google Test integrates well with other tools and libraries in the Google Test ecosystem. For example, it can be seamlessly used with Google Mock for creating mock objects and behaviours. This integration enhances the testing capabilities and flexibility.
+3. **Large and Active Community**: Google Test has a large and active community of developers. This means that there is ample support available in terms of documentation, tutorials, forums, and online resources. 
+4. **Mature and Stable**: It is a mature and stable framework that has been used extensively in industry projects and open-source software. 
+5. **Wide Platform Support**: Google Test supports multiple platforms, including Windows, Linux, macOS, and various compilers. It is compatible with popular development environments and build systems, making it suitable for a wide range of C++ projects.
+6. **Flexible and Extensible**: Google Test provides flexibility in test organization and customization. It allows you to structure your tests using test cases and test suites. You can also define custom test fixtures and customise test execution and reporting. Additionally, Google Test can be extended with custom assertion macros and utilities to suit your specific testing needs.
+
+
+
+## 3. Writing Unit test using GoogleTest
+
+For this chapter, we will consider the same example `fibonacci.cpp` that we used in the previous chapter. Also, for understanding how to use GoogleTest, we will simply be converting all the tests that we wrote manually in the same file using GoogleTest framework.
+
+For using GoogleTest in your code, you need to follow the following steps in general.
+
+
+
+### 3.1. Adding the required header files
+
+The first step would be to add the required header files in your program. For Googletest, you would need to add the following line in your code.
+
+```cpp
+#include "gtest/gtest.h"
+```
+
+
+
+### 3.2. Create your tests
+
+The next step would be to to define your test. Googletest uses the following convention in naming various tests.
+
+```cpp
+TEST(TestSuiteName, TestName)
+{
+    // Test logic and assertions
+}
+```
+
+The different parts in the above cell have the following meaning
+
+- `TEST`: This is the macro provided by the Google Test framework to define a test case.
```

Comment:
If someone is doing this course, probably they should know already, but maybe define what a macro is (or add a link to where it is explained).

Reply:
Yes, they should. I have now added a link for macro in commit 62316af.